test/services/onlinetesting/OnlineTestExtensionServiceSpec.scala

```diff
@@ -14,66 +14,166 @@
+        when(mockProgressResponse.phase1TestsStarted).thenReturn(true)
+        when(mockProfile.expirationDate).thenReturn(InFiveHours)
+        when(mockOtRepository.updateGroupExpiryTime(eqTo(applicationId), any())).thenReturnAsync()
+        when(mockAppRepository.removeProgressStatuses(eqTo(applicationId), any())).thenReturnAsync()
 
-      verify(repository).updateExpiryTime(userId, expirationDate.plusDays(extraDays))
+        underTest.extendTestGroupExpiryTime(applicationId, threeExtraDays).futureValue(timeout(Span(1, Seconds))) mustBe (())
+
+        verify(mockOtRepository).updateGroupExpiryTime(eqTo(applicationId), eqTo(InFiveHours.plusDays(threeExtraDays)))
+        verify(mockAppRepository).removeProgressStatuses(eqTo(applicationId), eqTo(statusToRemoveWhenExpiryInMoreThanThreeDays))
+        verify(mockAuditService).logEventNoRequest(eqTo("NonExpiredTestsExtended"), eqTo(Map("applicationId" -> applicationId)))
+        verifyNoMoreInteractions(mockAuditService, mockDateFactory)
+      }
     }
+    "return a failed Future" when {
+      "the application status doesn't allow an extension" in new TestFixture {
+        when(mockAppRepository.findProgress(any())).thenReturnAsync(successfulProgressResponse)
+        when(mockOtRepository.getPhase1TestGroup(applicationId)).thenReturnAsync(successfulTestProfile)
+        whenReady(underTest.extendTestGroupExpiryTime(applicationId, twoExtraDays).failed) { e =>
+          e mustBe (invalidStatusError)
+        }
+
+        verify(mockAppRepository).findProgress(eqTo(applicationId))
+        verify(mockOtRepository).getPhase1TestGroup(eqTo(applicationId))
+        verifyNoMoreInteractions(mockAppRepository, mockOtRepository, mockAuditService, mockDateFactory)
+      }
+      "find progress fails" in new TestFixture {
+        when(mockAppRepository.findProgress(any())).thenReturn(Future.failed(genericError))
+        whenReady(underTest.extendTestGroupExpiryTime(applicationId, twoExtraDays).failed) { e =>
+          e mustBe (genericError)
+        }
+        verify(mockAppRepository).findProgress(eqTo(applicationId))
+        verifyNoMoreInteractions(mockAppRepository, mockOtRepository, mockAuditService, mockDateFactory)
+      }
+      "No test phase 1 profile is available" in new TestFixture {
+        when(mockAppRepository.findProgress(any())).thenReturnAsync(successfulProgressResponse)
+        when(mockOtRepository.getPhase1TestGroup(applicationId)).thenReturnAsync(None)
+        whenReady(underTest.extendTestGroupExpiryTime(applicationId, twoExtraDays).failed) { e =>
+          e mustBe (noTestProfileFoundError)
+        }
+        verify(mockAppRepository).findProgress(eqTo(applicationId))
+        verify(mockOtRepository).getPhase1TestGroup(eqTo(applicationId))
+        verifyNoMoreInteractions(mockAppRepository, mockOtRepository, mockAuditService, mockDateFactory)
+      }
+      "remove status return an error and no audit event is emitted" in new TestFixture {
```

Comment:
I think it's OK, audit events should only happen when state is changed - although we should probably emit a normal log to say something's gone wrong